### DISCUSSION TITLE
State Mangment

### DISCUSSION THREAD

**Discussion by mo-fouad** [nandorojo/solito#109] [Ideas]
Hello guys, 
From your perspective and tryouts, what would be the best solution for state management here for all apps? Or should I use different state management approach for web and mobile?

I have tried to use [Redux](https://react-redux.js.org/), [Context API](https://reactjs.org/docs/context.html), and [Jotai](https://jotai.org/)
 but still feels like there is something messing in the loop. What do you think?
 
 Thanks :)

**Comment by nandorojo** [comment]
> should I use different state management approach for web and mobile?

use the same one here, I don’t see a reason to make them different.

for your other questions — this is a personal preference thing. for data, i would use `react-query`. i like `zustand` for non-data related global state. and for local state, `useState` does the job. i’ve also heard about things like state machines but haven’t looked into them yet.

**Reply by mo-fouad** [reply]
Thanks a lot for your Super quick answer ^_^ 
From your perspective, what would be the best practice to store the Authentication-Info, overall the app?
I mean, with every refresh should I check if I still have a user ? by hitting the server with the token? and where to store that token :D

**Reply by nandorojo** [reply]
i use firebase and react native firebase for that. there are many open discussions for it, but i haven’t had time to make an example. i might be able to if someone submits a significant sponsorship, but otherwise i’ll just have to make an auth example when i have some time.

**Reply by nandorojo** [reply]
#87 might help
